mmeds/tests/unit/test.py

```diff
@@ -0,0 +1,144 @@
+from subprocess import run
+import sys
+import os
+
+from mmeds.authentication import add_user, remove_user
+from mmeds.database import upload_metadata, upload_otu
+
+import mmeds.config as fig
+import mmeds.secrets as sec
+
+'''
+- to run all the tests: python test.py
+- to run a specific set of test: python test.py test_name1 test_name2 etc
+  - possible test names: authentication, database, documents, spawn, tool, tools, util, validate
+- to run all tests with error log output: python test.py log
+  - gives output as wanted for Travis CI
+'''
+
+testing = True
+log = False
+
+def main():
+    global log
+    if 'log' in sys.argv:
+        log = True
+    tests = set_tests(sys.argv)
+    users_added = add_users(tests)
+    setup_tests(tests)
+    run_tests(tests)
+    remove_users(users_added)
+
+def set_tests(sys_args):
+    tests = sys.argv[1:]
+    if 'log' in tests:
+        tests.remove('log')
+    # if there are no specific tests passed, then run all of them
+    if len(tests) == 0:
+        tests = ['authentication', 'database', 'documents', 'spawn', 'tool', 'tools', 'util', 'validate']
+    return tests
+
+def add_users(tests):
+    # add users as needed
+    # users_added keeps track of the number of users added so they can all be removed at the end
+    users_added = 0 
+    if ('database' in tests) or ('documents' in tests) or ('spawn' in tests) or ('tool' in tests) or ('tools' in tests):
```

Comment:
I can't tell on github but it looks like this line might be over 120 characters. If so you can break it up, either by putting parentheses around the whole if statement or adding a backslash. Like 
`if x == y and\`
`y == z:`

also generally in python the style is minimal use of parentheses. So the parens around `('spawn' in tests)` are allowed but not necessary. Usually they are only used for order of operations or clarity. I don't think they are necessary her, or in similar situations elsewhere in this file.